utils/ccomp.ml

```diff
@@ -87,6 +87,8 @@ let compile_file name =
                   then (Config.ocamlopt_cflags, Config.ocamlopt_cppflags)
                   else (Config.ocamlc_cflags, Config.ocamlc_cppflags) in
               (String.concat " " [Config.c_compiler; cflags; cppflags]))
+         (match output with | None -> "" | Some o -> "-o " ^ o)
+         opt
```

Comment:
Have you checked that all the C compilers support the `-o` option? (Sun compiler, Intel compiler, MSVC, gcc, clang, pcc)